File: sandbox/exercism/c/2020/rna_transcription.c
/* RNA Transcription from https://exercism.io */

#include "rna_transcription.h"
#include <string.h>
#include <stdlib.h>

char *to_rna(const char *dna)
{
  char *rna = malloc(sizeof(char) * strlen(dna));

  for (int i = 0; dna[i] != '\0'; i++) {
    switch(dna[i]) {
      case 'G':
        rna[i] = 'C';
        break;
      case 'C':
        rna[i] = 'G';
        break;
      case 'T':
        rna[i] = 'A';
        break;
      case 'A':
        rna[i] = 'U';
        break;
      default:
        return NULL;
    }
  }

  return rna;
}

File: sandbox/exercism/go/2025/rna_transcription.go
// https://exercism.org/tracks/go/exercises/rna-transcription

package strand

func ToRNA(dna string) string {
	mapping := map[rune]string{
		'G': "C",
		'C': "G",
		'T': "A",
		'A': "U",
	}

	strands := ""

	for _, r := range dna {
		strands += mapping[r]
	}

	return strands
}
